bcda/cclf/cclf.go

```diff
@@ -427,43 +447,61 @@ func sortCCLFArchives(cclfMap *map[string]map[int][]*cclfFileMetadata, skipped *
 		}
 		_ = zipReader.Close()
 
-		metadata, err := getCCLFArchiveMetadata(info.Name())
-		metadata.filePath = path
-		metadata.deliveryDate = info.ModTime()
-		if err != nil {
-			// skipping files with a bad name.  An unknown file in this dir isn't a blocker
-			fmt.Printf("Unknown file found: %s.\n", metadata)
-			log.Errorf("Unknown file found: %s", metadata)
+		// validate the top level zipped folder
+		err = validateCCLFFolderName(info.Name()); if err != nil {
 			*skipped = *skipped + 1
-
-			deleteThreshold := time.Hour * time.Duration(utils.GetEnvInt("BCDA_ETL_FILE_ARCHIVE_THRESHOLD_HR", 72))
-			if metadata.deliveryDate.Add(deleteThreshold).Before(time.Now()) {
-				newpath := fmt.Sprintf("%s/%s", os.Getenv("PENDING_DELETION_DIR"), info.Name())
-				err = os.Rename(metadata.filePath, newpath)
-				if err != nil {
-					fmt.Printf("Error moving unknown file %s to pending deletion dir.\n", metadata)
-					err = fmt.Errorf("error moving unknown file %s to pending deletion dir", metadata)
-					log.Error(err)
-					return err
-				}
+			msg := fmt.Sprintf("Skipping CCLF archive: %s.", info.Name())
+			fmt.Println(msg)
+			log.Warn(msg)
+			err = checkDeliveryDate(path, info.ModTime())
+			if err != nil {
+				fmt.Printf("Error moving unknown file %s to pending deletion dir.\n", path)
+				err = fmt.Errorf("error moving unknown file %s to pending deletion dir", path)
+				log.Error(err)
+				return err
 			}
 			return nil
 		}
 
-		if (*cclfMap)[metadata.acoID] != nil {
-			if (*cclfMap)[metadata.acoID][metadata.perfYear] != nil {
-				(*cclfMap)[metadata.acoID][metadata.perfYear] = append((*cclfMap)[metadata.acoID][metadata.perfYear], &metadata)
+		for _, f := range zipReader.File {
+			metadata, err := getCCLFFileMetadata(f.Name)
+			metadata.filePath = path
+			metadata.deliveryDate = info.ModTime()
+
+			if err != nil {
+				// skipping files with a bad name.  An unknown file in this dir isn't a blocker
+				fmt.Printf("Unknown file found: %s.\n", f.Name)
```

Comment:
Do we have to log this?  We're going to get TONS of these log statements due to skipping CCLF3, 4, 5, 6, ... etc.

Reply:
we don't have to log this if we don't think we need it